dissect/target/filesystems/vmfs.py

```diff
@@ -92,18 +87,20 @@ def scandir(self) -> Iterator[FilesystemEntry]:
             path = fsutil.join(self.path, f.name, alt_separator=self.fs.alt_separator)
             yield VmfsFilesystemEntry(self.fs, path, f)
 
-    def is_dir(self) -> bool:
-        """Return whether this entry is a directory. Resolves symlinks when possible."""
+    def is_dir(self, follow_symlinks: bool = True) -> bool:
+        """Return whether this entry is a directory."""
         try:
-            return self._resolve().entry.is_dir()
+            return self._resolve(follow_symlinks=follow_symlinks).entry.is_dir()
         except FilesystemError:
             return False
 
-    def is_file(self) -> bool:
-        """Return whether this entry is a file. Resolves symlinks when possible."""
+    def is_file(self, follow_symlinks: bool = True) -> bool:
+        """Return whether this entry is a file."""
         try:
-            resolved = self._resolve()
-            return resolved.entry.is_file() or resolved.entry.is_system()
+            return (
+                self._resolve(follow_symlinks=follow_symlinks).entry.is_file()
```

Comment:
Can you cache the result of `self._resolve(follow_symlinks=follow_symlinks)` in a variable?

Reply:
It was never cached before. I'd rather not make that type of changes for this PR.